app/src/main/java/tech/ula/ServerService.kt

```diff
@@ -27,6 +30,9 @@ class ServerService : Service() {
 
     private lateinit var broadcaster: LocalBroadcastManager
 
+    private var wakeLock: PowerManager.WakeLock? = null
```

Comment:
1 - Good suggestion. Will look into this.
2 - I tried that way but I got a warning that it could cause a potential memory leak and should use application context instead.
2 - Will do but according to the documentation, the tag should be delimited by a colon.

"use a unique prefix delimited by a colon for your app/library (e.g. gmail:mytag) to make it easier to understand where the wake locks comes from. This namespace will also avoid collision for tags inside your app coming from different libraries which will make debugging easier."

So, maybe "Ula:ServerService"?

Reply:
2. If it's warning about memory leaks you can leave `applicationContext` as is. 
3. That seems like a good choice.